packages/agent/src/oidc.ts

```diff
@@ -0,0 +1,703 @@
+  scope: string;
+
+  /** The type of response desired (e.g. `id_token`) */
+  response_type: string;
+
+  /** the URL to which the Identity Provider will post the Authorization Response */
+  redirect_uri: string;
+
+  /** The URI to which the SIOPv2 Authorization Response will be sent (Tim's note: not used with encrypted request JWT)*/
+  response_uri?: string;
+
+  /**
+   * An opaque value used to maintain state between the request and the callback.
+   * Recommended for security to prevent CSRF attacks.
+   */
+  state: string;
+
+  /**
+   * A string value used to associate a client session with an ID token to mitigate replay attacks.
+   * Recommended when requesting ID tokens.
+   */
+  nonce: string;
+
+  /**
+   * The PKCE code challenge.
+   * Required if `code_challenge_method` is used. Enhances security for public clients (e.g., single-page apps,
+   * mobile apps) by requiring an additional verification step during token exchange.
+   */
+  code_challenge: string;
+
+  /** The method used for the PKCE challenge (typically `S256`). Must be present if `code_challenge` is included. */
+  code_challenge_method: 'S256';
+
+  /**
+   * An ID token previously issued to the client, passed as a hint about the end-user’s current or past authenticated
+   * session with the client. Can streamline user experience if already logged in.
+   */
+  id_token_hint?: string;
+
+  /** A hint to the authorization server about the login identifier the user might use. Useful for pre-filling login information. */
+  login_hint?: string;
+
+  /** Requested Authentication Context Class Reference values. Specifies the authentication context requirements. */
+  acr_values?: string;
+
+  /** When using a PAR for secure cross device flows we use a "form_post" rather than a "direct_post" */
+  response_mode: 'direct_post';
+
+  /** Used by PFI to request VCs as input to IDV process. If present, `response_type: "vp_token""` MUST also be present */
+  presentation_definition?: any;
+
+  /** A JSON object containing the Verifier metadata values (Tim's note: from TBD KCC Repo) */
+  client_metadata?: {
+    /** Array of strings, each a DID method supported for the subject of ID Token	*/
+    subject_syntax_types_supported: string[];
+    /** Human-readable string name of the client to be presented to the end-user during authorization */
+    client_name?: string;
+    /** URI of a web page providing information about the client */
+    client_uri?: string;
+    /** URI of an image logo for the client */
+    logo_uri?: string;
+    /** Array of strings representing ways to contact people responsible for this client, typically email addresses */
+    contacts?: string[];
+    /** URI that points to a terms of service document for the client */
+    tos_uri?: string;
+    /** URI that points to a privacy policy document */
+    policy_uri?: string;
+  };
+};
+
+/**
+ * An auth request that is compatible with both Web5 Connect and (hopefully, WIP) OIDC SIOPv2
```

Comment:
Sanity: remove `(hopefully, WIP)`? Same for `Web5ConnectAuthResponse`.

Reply:
this hasn't been validated with pure OIDC we had this same conversation about a different comment that you were asking me to remove. when we do both connect and OIDC (w/o connect) in the wallet I will remove this